Adyen/Core/Components/Base/Component.swift

```diff
@@ -7,16 +7,7 @@
 import Foundation
 
 /// A component provides payment method-specific UI and handling.
-public protocol Component: AnyObject {
-    
-    /// Defines the environment used to make networking requests.
-    var environment: Environment { get set }
-    
-    /// The client key that corresponds to the webservice user you will use for initiating the payment.
-    /// See https://docs.adyen.com/user-management/client-side-authentication for more information.
-    var clientKey: String? { get set }
-    
-}
+public protocol Component: APIContextAware { }
```

Comment:
We no longer pass-through `environment` and `clientKey` to every generic component, right?

For example before it was:
```swift
(item as? Component).environment = theEnvironment 
```

Now *environment* is read-only `{ get }` and we setting it via `init`

---

How about we only apply `APIContextAware` to specific component's that actually needs it?

Reply:
I think we talked about this previously. We want to make this future-proof in case we need to change implementation of a component to use any internal resources. We won't have to change the interface in that case.